dev/docker-compose.yaml

```diff
@@ -46,71 +43,111 @@ services:
-      - ../apisix/apisix.yaml:/usr/local/apisix/conf/apisix.yaml:ro # for hot-reloading
+      - ./apisix_config.yaml:/usr/local/apisix/conf/config.yaml:ro
+      - ../apisix/apisix.yaml:/usr/local/apisix/conf/apisix.yaml:ro
     environment:
+      NO_DAPR: true
       ORY_WEBHOOK_BASIC_AUTH: '{"username":"ory","password":"2341"}'
     ports:
       - "9080:9080"
 
-  task-svc:
-    build:
-      context: ..
-      dockerfile: Dockerfile.standalone
-      args:
-        - SERVICE=task-svc
-    profiles:
-      - full
-    volumes:
-      - ./dapr.yaml:/app/dapr/dapr.yaml:ro
+  # dapr sidecar for apisix
+  apisix-dapr:
+    image: "daprio/daprd:1.12.4"
     restart: unless-stopped
-    environment:
-      INSECURE_FAKE_TOKEN_ENABLE: true
-      POSTGRES_DSN: postgres://postgres:postgres@postgres/task-svc
-      SECRETSTORE_REDIS_HOST: "redis:6379"
-      SECRETSTORE_REDIS_PASSWORD: ""
-      SECRETSTORE_REDIS_USERNAME: ""
+    depends_on:
+      - apisix
+    volumes:
+      - ./dapr.yaml:/dapr.yaml:ro
+    command: [
+      "./daprd",
+      "--app-id", "gateway",
+      "--dapr-http-read-buffer-size", "8",
+      "--config", "/dapr.yaml",
+      "--log-level", "debug",
+      "--app-port", "9080",
+      "--dapr-internal-grpc-port", "35001"
+    ]
+    network_mode: "service:apisix"
 
-  user-svc:
+  # services compiles and runs all services using dapr_run.yaml
+  # also runs initial migrations for all services
+  services:
     build:
-      context: ..
-      dockerfile: Dockerfile.standalone
+      context: ../images/dev-go-custom
       args:
-        - SERVICE=user-svc
-    profiles:
-      - full
+        # renovate: dev-tag
+        TAG: '1'
+        UID: ${UID:-1000}
+        GID: ${GID:-1000}
     volumes:
-      - ./dapr.yaml:/app/dapr/dapr.yaml:ro
-    restart: unless-stopped
+      # we currently don't mount (i.e. persist) the go module cache
+      # this increases consistency in builds and dependencies, in trade for increased initial startup time
+
+      # mounts the whole repo
+      - ../:/helpwave
+    working_dir: /helpwave
+    user: "helpwave"
+    depends_on:
+      postgres:
+        condition: service_healthy
+      consul:
+        condition: service_started
     environment:
-      INSECURE_FAKE_TOKEN_ENABLE: true
-      POSTGRES_DSN: postgres://postgres:postgres@postgres/user-svc
       SECRETSTORE_REDIS_HOST: "redis:6379"
-      SECRETSTORE_REDIS_PASSWORD: ""
-      SECRETSTORE_REDIS_USERNAME: ""
+    command:
+      - /usr/bin/bash
+      - -c
+      - |
+        bash migrate.sh
+        dapr run --run-file dev/dapr_run.yaml --config dev/dapr.yaml
+    restart: unless-stopped
+    network_mode: "service:devcontainer" # share ports with devcontainer
 
-  ory-svc:
+  # devcontainer is where the developer should do their development in
+  devcontainer:
     build:
-      context: ..
-      dockerfile: Dockerfile.standalone
+      context: ../images/dev-go-custom
```

Comment:
Could it be more beneficial to use an officially maintained devcontainer image for go that build your own?

* https://hub.docker.com/_/microsoft-devcontainers
* https://hub.docker.com/_/microsoft-vscode-devcontainers

Reply:
From what I can gather, these images are not meant to be run standalone.